### PR TITLE
Add check if database is not connected when get method to retrieve alembic version from table is used.

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Screenshot from 2021-02-01 11-44-06

None in revision should never appear

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: thoth/metrics_exporter/jobs/db.py [hunk 1]
```diff
@@ -119,9 +119,16 @@ def set_script_head_revision(cls):
...
     def set_table_head_revision(cls):
         """Set metric for indicating database revision exposed by alembic table."""
-        metrics.database_schema_revision_table.labels(
-            "metrics-exporter", cls.graph().get_table_alembic_version_head(), Configuration.DEPLOYMENT_NAME
-        ).set(1)
+        table_revision_head = None
+        try:
+            table_revision_head = cls.graph().get_table_alembic_version_head()
+        except Exception as e:
+            _LOGGER.warning("Database revision table could not be retrieved: %r", e)
```
Comment (fridex) [inline thoth/metrics_exporter/jobs/db.py:126]: Maybe an exception should be reported (but I don't know the issue behind this):

```suggestion
            _LOGGER.exception("Database revision table could not be retrieved: %r", e)
```

**Review by fridex — COMMENTED** [review]
Code looks good to me, it's not clear to me what the issue is in this case. Could you elaborate, please?

**Comment by pacospace** [comment]
> Code looks good to me, it's not clear to me what the issue is in this case. Could you elaborate, please?

When using `cls.graph().get_table_alembic_version_head()` we should not receive `None` revision. Not sure if it is an issue in the implementation method, that sometimes does not read the current version from database correctly or metric-exporters was not able to talk to the database.

Unfortunately, we cannot see metrics past 12 hours so we cannot see when this happened, maybe after initial deployment of new metrics exporter or randomly at some point and in that case also other metrics would not have been reachable.

**Comment by fridex** [comment]
> > Code looks good to me, it's not clear to me what the issue is in this case. Could you elaborate, please?
> 
> When using `cls.graph().get_table_alembic_version_head()` we should not receive `None` revision. Not sure if it is an issue in the implementation method, that sometimes does not read the current version from database correctly or metric-exporters was not able to talk to the database.

Not sure if silently passing the exception on the metrics-exporter side would fix this, if I understand this correctly. We should keep exceptions being reported so we know what's happening in the cluster. Could the issue be caused by the default value of the metric on metrics-exporter startup? If `THOTH_METRICS_EXPORTER_WEAK_READINESS` is set to 1, metrics-exporter can possibly report the default value as it did not scrape all the metrics yet. In which deployment you spotted this?

**Review by fridex — APPROVED** [review]
Ah, ok. Now I see your point and the issue. Sorry for confusion :+1:

**Comment by pacospace** [comment]
> > > Code looks good to me, it's not clear to me what the issue is in this case. Could you elaborate, please?
> > 
> > 
> > When using `cls.graph().get_table_alembic_version_head()` we should not receive `None` revision. Not sure if it is an issue in the implementation method, that sometimes does not read the current version from database correctly or metric-exporters was not able to talk to the database.
> 
> Not sure if silently passing the exception on the metrics-exporter side would fix this, if I understand this correctly. We should keep exceptions being reported so we know what's happening in the cluster. Could the issue be caused by the default value of the metric on metrics-exporter startup? If `THOTH_METRICS_EXPORTER_WEAK_READINESS` is set to 1, metrics-exporter can possibly report the default value as it did not scrape all the metrics yet. In which deployment you spotted this?

Stage environment.

**Pull request merged**